vim/vscode-neovim.vim

```diff
@@ -58,7 +58,13 @@ function! VSCodeCallVisual(cmd, leaveSelection, ...) abort
     elseif mode ==# 'v' || mode ==# "\<C-v>"
         let startPos = getpos('v')
         let endPos = getpos('.')
-        call VSCodeCallRangePos(a:cmd, startPos[1], endPos[1], startPos[2], endPos[2] + 1, a:leaveSelection, a:000)
+        "  Check if the cursor is at the beginning of the visual selection
+        if (startPos[1] == endPos[1] && startPos[2] > endPos[2]) || startPos[1] > endPos[1]
+            let startPos[2] = startPos[2] + 1
+        else
+            let endPos[2] = endPos[2] + 1
+        endif
```

Comment:
since this is used in multiple places, it's worth extracting it to a function like `fixVisualPos()`. That gives a hint to future readers that happen to look at this function. Also helps if a bug fix is needed. 

Is it possible to add a test?

Reply:
@justinmk, I've extracted duplicated logic into the `s:fixVisualPos` function and added a test.